debian.Dockerfile

```diff
@@ -0,0 +1,42 @@
+# Setup build arguments with default versions
+ARG AWS_CLI_VERSION=1.16.166
+ARG TERRAFORM_VERSION=0.11.14
+
+# Download Terraform binary
+FROM debian:stretch-20190506-slim as terraform
+ARG TERRAFORM_VERSION
+RUN apt-get update
+RUN apt-get install -y curl=7.52.1-5+deb9u9
+RUN apt-get install -y unzip=6.0-21+deb9u1
+RUN apt-get install -y gnupg=2.1.18-8~deb9u4
+RUN curl -Os https://releases.hashicorp.com/terraform/${TERRAFORM_VERSION}/terraform_${TERRAFORM_VERSION}_SHA256SUMS
+RUN curl -Os https://releases.hashicorp.com/terraform/${TERRAFORM_VERSION}/terraform_${TERRAFORM_VERSION}_linux_amd64.zip
+RUN curl -Os https://releases.hashicorp.com/terraform/${TERRAFORM_VERSION}/terraform_${TERRAFORM_VERSION}_SHA256SUMS.sig
+COPY hashicorp.asc hashicorp.asc
+RUN gpg --import hashicorp.asc
+RUN gpg --verify terraform_${TERRAFORM_VERSION}_SHA256SUMS.sig terraform_${TERRAFORM_VERSION}_SHA256SUMS
+RUN sha256sum -c --ignore-missing terraform_${TERRAFORM_VERSION}_SHA256SUMS
+RUN unzip -j terraform_${TERRAFORM_VERSION}_linux_amd64.zip
+
+# Install AWS CLI using PIP
+FROM debian:stretch-20190506-slim as aws-cli
+ARG AWS_CLI_VERSION
+RUN apt-get update
+RUN apt-get install -y python3=3.5.3-1
+RUN apt-get install -y python3-pip=9.0.1-2+deb9u1
+RUN pip3 install awscli==${AWS_CLI_VERSION}
+
+# Build final image
+FROM debian:stretch-20190506-slim
+RUN apt-get update --no-install-recommends \
+  # TODO: Handle potential download issue when adding multiples packages with APT
+  && apt-get install -y python3=3.5.3-1 ca-certificates=20161130+nmu1+deb9u1 \
```

Comment:
peux tu ajouter jq et terragrunt stp ?

Reply:
Je ne suis pas certains que ce soit le meilleur endroit pour ajouter ces outils.

Cette image se veux simple et ne porte que les CLI Terraform et le AWS, il me semble donc qu'il faudrait plutôt créer une autre image qui étend de celle-ci pour en étendre les fonctionnalités (typiquement pour des workshop avec des besoins spécifiques), qu'en penses-tu @gmembre-zenika  ?

Reply:
la sortie de la ligne de commande aws est du json, pour la parser correctement, jq est LA solution. Les 2 sont, pour moi, inséparables.

Reply:
C'est noté pour jq, je l'ajoute !

Quel est le cas d'usage précis de Terragrunt ? Je comprends bien son utilité mais c'est un choix de workflow qui ne me semble pas spécifique au scope de cette image Docker.

Reply:
merci pour jq.

terragrunt est un outil qui vient en surcouche à terraform pour améliorer la modularisation de ton projet. Je te montre demain stv :)
L'ajouter serait un plus mais il n'est pas indispensable. Ca peut etre fait plus tard.

Reply:
Oui il est très utile, je suis en phase pour choisir ou non de l'ajouter dans un second temps 😉